modules/phase_field/src/kernels/NestKKSACBulkC.C

```diff
@@ -0,0 +1,151 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "NestKKSACBulkC.h"
+
+registerMooseObject("PhaseFieldApp", NestKKSACBulkC);
+
+InputParameters
+NestKKSACBulkC::validParams()
+{
+  InputParameters params = KKSACBulkBase::validParams();
+  params.addClassDescription("KKS model kernel (part 2 of 2) for the Bulk Allen-Cahn. This "
+                             "includes all terms dependent on chemical potential.");
+  params.addRequiredCoupledVar("global_cs", "The interpolated concentrations c, b, etc");
+  params.addRequiredParam<std::vector<MaterialPropertyName>>(
+      "ci_names",
+      "Phase concentrations. The order must match Fa, Fb, and global_c, for example, c1, "
+      "c2, b1, b2, etc");
```

Comment:
we can leave this as is for now, but we do have a `std::vector<std::vector<T>>`  parameter type (user input would look like `'c1 c2; b1 b2'`

Reply:
With `std::vector<std::vector<T>>` and user input of ci_names `c1 c2; b1 b2`, do we refer to `b1` as `_ci_names[1][0]`? I would probably leave it as is so we don't need to change all the for() loops.